ridley/src/System/Metrics/Prometheus/Ridley/Types.hs

```diff
@@ -53,6 +54,16 @@ data RidleyMetricHandler = forall c. RidleyMetricHandler {
   , updateMetric :: c -> Bool -> IO ()
   , flush        :: !Bool
   -- ^Whether or net to flush this Metric
+  , label        :: T.Text
+  }
+
+mkRidleyMetricHandler :: forall c. HasCallStack
+                      => c -> (c -> Bool -> IO ()) -> Bool -> RidleyMetricHandler
+mkRidleyMetricHandler c runC flsh = withFrozenCallStack $ RidleyMetricHandler {
+    metric       = c
+  , updateMetric = runC
+  , flush        = flsh
+  , label        = T.pack $ prettyCallStack $ popCallStack callStack
```

Comment:
Good idea to add a smart constructor.

Including the call stack is  a nice trick, but I think it might be better if it were passed separately from the label? Then you could have a message like
```
Couldn't update handler for my-crashful due to user error (CRASH!)
CallStack (from HasCallStack): ...
```
You could also do with adding `HasCallStack` constraints to the functions in this library that call `mkRidleyMetricHandler`, assuming you want the call stack to point to application code.